lib/galaxy-social.py

```diff
@@ -0,0 +1,152 @@
+import fnmatch
+import importlib
+import json
+import os
+import sys
+
+import jsonschema
+import markdown
+import requests
+import yaml
+from bs4 import BeautifulSoup
+from github_comment import comment_to_github
+
+with open("plugins.yml", "r") as file:
+    plugins_config = yaml.safe_load(file)
+
+plugins = {}
+for plugin in plugins_config["plugins"]:
+    if "preview" in sys.argv and plugin["name"].lower() != "markdown":
+        continue
+
+    if plugin["enabled"]:
+        module_name, class_name = plugin["class"].rsplit(".", 1)
+
+        try:
+            module = importlib.import_module(f"plugins.{module_name}")
+            plugin_class = getattr(module, class_name)
+        except:
+            comment_to_github(
+                f"Error with plugin {module_name}.{class_name}.", error=True
+            )
+
+        try:
+            config = {
+                key: os.environ.get(value)
+                for key, value in plugin["config"].items()
+                if (not isinstance(value, int) and os.environ.get(value) is not None)
```

Comment:
This doesn't seem to make much sense.
1. There are values in the yml file (like the base_url and the save_path), which aren't integers and should still not trigger env replacements; some access tokens, provided a user puts them in the yml directly, would also be hex strings I assume?
2. Currently, if the yml value *is* an int, it just gets skipped

A simpler way would be:
1. populate the dict with the yml values first
2. come up with a speical way to write env placeholders (.e.g. surrounding double underscores or something) and try to replace only values in the config dict that look like env replacement patterns.

Reply:
It now processes the env variables if they are started with `$`.